pkg/tests/discovery_selectors_test.go

```diff
@@ -0,0 +1,93 @@
+// Copyright 2021 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ossm
+
+import (
+	"testing"
+
+	"github.com/maistra/maistra-test-tool/pkg/app"
+	"github.com/maistra/maistra-test-tool/pkg/tests/ossm"
+	"github.com/maistra/maistra-test-tool/pkg/util/check/assert"
+	"github.com/maistra/maistra-test-tool/pkg/util/oc"
+	"github.com/maistra/maistra-test-tool/pkg/util/pod"
+	"github.com/maistra/maistra-test-tool/pkg/util/retry"
+	. "github.com/maistra/maistra-test-tool/pkg/util/test"
+)
+
+func TestConversionDiscoverySelectors(t *testing.T) {
+	NewTest(t).Id("T41").Groups(Full, InterOp).Run(func(t TestHelper) {
+		t.Log("This test checks for discoverySelectors to convert from helm values to smcp spec")
+
+		ns := "bookinfo"
+
+		ossm.DeployControlPlane(t)
+
+		t.NewSubTest("discoverySelecotrs").Run(func(t TestHelper) {
+			t.LogStep("Update SMCP spec.controlPlane.meshConfig.discoverySelectors: Enabled")
+			oc.Patch(t, meshNamespace, "smcp", smcpName, "merge", `{"spec":{"controlPlane":{"disoverySelectors"}}}}}`)
```

Comment:
1. This patch is malformed.
2. Why do you patch SMCP if you apply `DSEnabled` that has `discoverySelectors` specified?

Reply:
Regarding 2: I think we should keep the patch for now until we implement a standard way of doing this across all the tests. Otherwise, we'll end up with every test using its own way of deploying the custom SMCP, which will take more effort to standardize. Also, be aware that `oc apply` does not properly update the SMCP, so doing it in each test might cause additional problems that we currently don't have (`ossm.DeployControlPlane()` takes care of this by using `oc.ReplaceOrApply()`).